core/lib/spree/permission_sets/default_customer.rb

```diff
@@ -8,7 +8,7 @@ def activate!
         can :display, OptionType
         can :display, OptionValue
         can :create, Order
-        can [:read, :update], Order do |order, token|
+        can [:read, :update], Order, Order.where(user: user) do |order, token|
```

Comment:
I think we need more of these. The third parameter is for scopes, the block for single records. Then we should be able to still support cancancan 2

Reply:
@tvdeyen I'm actually surprised it worked before, since we didn't have a scope. Anyway, this seems to be enough to make it work.

Reply:
:man_shrugging:

Reply:
@coorasse if you have some time, can you take a quick look here, please? Just to be sure we are not missing something important. Thanks!